darwin/dataset/remote_dataset.py

```diff
@@ -73,9 +94,20 @@ def pull(self):
 
             z.extractall(annotations_dir)
             annotation_format = "json"
-            return download_all_images_from_annotations(
+            progress, count = download_all_images_from_annotations(
                 self._client.url, annotations_dir, images_dir, annotation_format
             )
+            # If blocking is selected, download the dataset on the file system
+            if blocking:
+                if multi_threaded:
+                    import multiprocessing as mp
+                    with mp.Pool(mp.cpu_count()) as pool:
+                        pool.map(RemoteDataset._f, progress())
+                else:
+                    for f in progress():
+                        f()
```

Comment:
Perhaps this would be cleaner? (Line 110 would also need to be removed if this suggestion is accepted)
```suggestion
                        f()
                return None, count
            return progress
```